### DISCUSSION TITLE
Updating Gradle or Gradle Wrapper

### DISCUSSION THREAD

**Discussion by mcraghead** [apache/netbeans#7046] [Q&A]
I'm new to NetBeans and to Gradle. I'm using JDK 21 and it seems that Gradle 8.4, which is part of Netbeans 20, is not compatible. How do I update Gradle/Gradle wrapper to the most recent, stable version?

Also, I sense that there is a distinction between Gradle and Gradle wrapper. Is that correct? If so, what is the difference?

Thank you.

**Comment by neilcsmith-net** [comment]
Amongst other ways, right click on the project, select `Run Gradle > Tasks` and then in the dialog enter ...

```
wrapper --gradle-version latest
```

... and run it.

After doing this on one project, you might find that new projects pick this up automatically.  NB21 has some fixes to work better here, from 21-rc3 (keep an eye on #6962 )

Gradle wrapper is a small script that exists in your project to download and run the configured version of Gradle needed to build / run your project.  NetBeans also supports the Maven wrapper that does similar.

Neither Gradle or the wrapper are strictly part of NetBeans - the Gradle Tooling API is.  A library for IDEs and other tools to run Gradle projects.  Yet another piece of the puzzle! :smile:
